src/Dapr.Client/PublishHttpClient.cs

```diff
@@ -0,0 +1,95 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+namespace Dapr
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text.Json;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using static Dapr.DaprUris;
+
+    /// <summary>
+    /// A client for interacting with the Dapr publish endpoints using <see cref="HttpClient" />.
+    /// </summary>
+    public class PublishHttpClient : PublishClient
+    {
+        private readonly HttpClient client;
+        private readonly JsonSerializerOptions serializerOptions;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PublishHttpClient"/> class.
+        /// </summary>
+        /// <param name="client">The <see cref="HttpClient" />.</param>
+        /// <param name="serializerOptions">The <see cref="JsonSerializerOptions" />.</param>
+        public PublishHttpClient(HttpClient client, JsonSerializerOptions serializerOptions = null)
+        {
+            if (client is null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            this.client = client;
+            this.serializerOptions = serializerOptions;
+        }
+
+        /// <inheritdoc/>
+        public override async Task PublishEventAsync<TRequest>(string topicName, TRequest publishContent, CancellationToken cancellationToken = default)
+        {
+            if (string.IsNullOrEmpty(topicName))
+            {
+                throw new ArgumentNullException("The value cannot be null or empty", nameof(topicName));
+            }
+
+            if (publishContent is null)
+            {
+                throw new ArgumentNullException("The value cannot be null or empty", nameof(publishContent));
```

Comment:
Can remove the errors message here and just use the default since it's a simple null check.